src/machine.rs

```diff
@@ -918,16 +918,16 @@ impl OlmMachine {
         let me = self.inner.clone();
 
         future_to_promise(async move {
-            let request = me.backup_machine().backup().await?;
-
-            let Some(request) = request else {
-                return Ok(None);
-            };
+            match me.backup_machine().backup().await? {
+                Some((transaction_id, keys_backup_request)) => {
+                    Ok(Some(JsValue::from(requests::KeysBackupRequest::try_from((
```

Comment:
not sure we need the explicit `JsValue::from` here? I think `future_to_promise` will work with anything which implements `Into<JsValue>` (as `KeybackupRequest does`) ?

Reply:
Ah that's nice, I'll get rid of it then.